### ISSUE TITLE
perf (www): Explicitly define GraphQL data types for gatsbyjs.org

### ISSUE THREAD

**Issue by tesseralis** [gatsbyjs/gatsby#24913]
## Description

[Explicitly define](https://www.gatsbyjs.org/docs/schema-customization/#explicitly-defining-data-types) the GraphQL data types used in gatsbyjs.org:

* MDX frontmatter used in the docs and blog
* YAML files such as `StartersYaml`, `SitesYaml`, `CreatorsYaml`, etc.
* `GatsbyApiCall`
* `gatsby-transformer-documentationjs` #24592 
* `gatsby-screenshot-placeholder`

## Motivation

The gatsby website has a lot of pieces in it and is really slow to build. There are minor ways to make it faster, such as using [`GATSBY_SCREENSHOT_PLACEHOLDER`](https://github.com/gatsbyjs/gatsby/tree/master/www#running-slow-build-screenshots-placeholder), but for other than that, there's no way to make the build faster.

One possibility is to remove various sources from the app — for example, being able to run the site while not sourcing from the blog. But this causes GraphQL build issues:

```
Cannot query field "author" on type "MdxFrontmatter".
Cannot query field "twittercard" on type "MdxFrontmatter".
Cannot query field "showImageInArticle" on type "MdxFrontmatter".
```

This is especially important for internationalization of gatsbyjs.org — we need translators and translation maintainers to be able to accessibly compile and run localized versions of the gatsby site. Currently the localized sites don't even run because the website theme depends on the gatsby source code, which the translations don’t have implemented yet.

**Comment by tesseralis** [comment]
Resolved by #24943, #24944 among others

**Comment by AishaBlake** [comment]
I'm closing this issue for now because an internal team is working on some website changes that will affect our approach. We'll come back to this work in the near future!
